rustls/tests/api.rs

```diff
@@ -2936,6 +2936,64 @@ fn sni_resolver_rejects_bad_certs() {
     );
 }
 
+/// Represents a SigningKey that returns None for its SPKI.
+#[derive(Debug)]
+struct SigningKeyNoneSpki;
+
+impl sign::SigningKey for SigningKeyNoneSpki {
+    fn choose_scheme(
+        &self,
+        _offered: &[SignatureScheme],
+    ) -> Option<Box<dyn sign::Signer>> {
+        unimplemented!("Not meant to be called during tests")
+    }
+
+    fn algorithm(&self) -> rustls::SignatureAlgorithm {
+        unimplemented!("Not meant to be called during tests")
+    }
+}
+
+/// Represents a SigningKey that returns Some for its SPKI.
+#[derive(Debug)]
+struct SigningKeySomeSpki;
+
+#[cfg(test)]
+impl sign::SigningKey for SigningKeySomeSpki {
+    fn public_key(&self) -> Option<pki_types::SubjectPublicKeyInfoDer> {
+        let chain = KeyType::Rsa2048.get_chain();
+        let cert = ParsedCertificate::try_from(chain.first().unwrap()).unwrap();
+        let spki = cert.subject_public_key_info().into_owned();
+        Some(spki)
+    }
+
+    fn choose_scheme(
+        &self,
+        _offered: &[SignatureScheme],
+    ) -> Option<Box<dyn sign::Signer>> {
+        unimplemented!("Not meant to be called during tests")
+    }
+
+    fn algorithm(&self) -> rustls::SignatureAlgorithm {
+        unimplemented!("Not meant to be called during tests")
+    }
+}
+
+#[test]
+#[cfg(test)]
+fn test_keys_match() {
```

Comment:
I think the style guide ordering would prescribe this test being _before_ the `SigningKeyNoneSpki` and `SigningKeySomeSpki` impls since it relies on them.

Reply:
Done!